flow/hooks/log_operations.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2018 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Builtin execution hook for basic logging."""
+import logging
+
+
+class LogOperations:
+    """:class:`~.LogOperations` logs information about the execution of operations to a log file.
+
+    This hooks provides information, optionally, on the start, successful completion, and/or
+    erroring of one or more operations in a `flow.FlowProject` instance. The logs are stored in a
+    file given by the parameter ``fn_logfile``. This file will be appended to if it already exists.
+
+    The default formating for the log provides the time, job id, log level, and log message.
+
+    .. note::
+
+        All logging is performed at the INFO level. To ensure outputs are captured in log files,
+        use the `--debug` flag when running or submitting jobs, or specify
+        `submit_options=--debug` in your directives (example shown below).
+
+
+    Examples
+    --------
+    The following example will install :class:`~.LogOperations` at the operation level.
+    Where the log will be stored in a file name `foo.log` in the job workspace.
+
+    .. code-block:: python
+        from flow import FlowProject
+        from flow.hooks import LogOperations
+
+
+        class Project(FlowProject):
+            pass
+
+
+        def install_operation_log_hook(operation_name, project_cls):
+            log = LogOperation(f"{operation_name}.log")
+            return lambda op: log.install_operation_hooks(op, project_cls)
+
+
+        @install_operation_log_hook("foo", Project)
+        @Project.operation(directives={
+            "submit_options": "--debug"  # Always submit operation foo with the --debug flag
+        })
+        def foo(job):
+            pass
+
+
+    The code block below provides an example of how install  :class:`~.LogOperations` to a
+    instance of :class:`~.FlowProject`
+
+    .. code-block:: python
+        from flow import FlowProject
+        from flow.hooks import LogOperations  # Import build
+
+
+        class Project(FlowProject):
+            pass
+
+
+        # Do something
+
+
+        if __name__ == "__main__":
+            project = Project()
+            project = LogOperations().install_project_hooks(project)
+            project.main()
+
+
+
+    Parameters
+    ----------
+    fn_logfile: log filename
```

Comment:
The format is `parameter_name : type`.
```suggestion
    fn_logfile : str
```